### PR TITLE
Add the ability to migrate formats from the the govuk index

### PR DESCRIPTION
**Pull request by dwhenry**
This will support the migration to using the publishing API as the source of
data for search. This will allow the documents to be migrated across a format
at a time.

Please note that the implementation used here is not compatible with ES 2.4
https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-indices-filter.html
and will need to be updated separately there.

https://trello.com/c/J5DkD5Rn/230-allow-individual-formats-to-be-switched-over-to-use-the-new-govuk-index

### CODE REVIEW

**Comment by MatMoore** [comment]
Only comment I have now is this should go around the `filters` bit since it shouldn't influence the score in any way.

File: test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require 'integration_test_helper'
+
+class GovukIndex::PublishingEventProcessorTest < IntegrationTest
+  def test_defaults_to_excluding_govuk_index_records
+    insert_document('mainstream_test', title: 'mainstream answer', link: '/mainstream/answer', format: 'answer')
+    insert_document('mainstream_test', title: 'mainstream help', link: '/mainstream/help', format: 'help_pages')
+    insert_document('govuk_test', title: 'govuk help', link: '/govuk/help', format: 'help_pages')
```
Comment (brenetic) [inline test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb]: ~~help_pages~~ => `help_page`

File: test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    get "/search"
+
+    assert_equal(['mainstream answer', 'mainstream help'], parsed_response['results'].map { |r| r['title']})
+  end
+
+  def test_can_enable_format_to_use_govuk_index
+    insert_document('mainstream_test', title: 'mainstream answer', link: '/mainstream/answer', format: 'answer')
+    insert_document('mainstream_test', title: 'mainstream help', link: '/mainstream/help', format: 'help_pages')
+    insert_document('govuk_test', title: 'govuk help', link: '/govuk/help', format: 'help_pages')
```
Comment (brenetic) [inline test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb]: ~~help_pages~~ => `help_page`

**Review by brenetic — COMMENTED** [review]
Couple of comments. Otherwise approach looks good 👍

File: test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    insert_document('mainstream_test', title: 'mainstream help', link: '/mainstream/help', format: 'help_page')
+    insert_document('govuk_test', title: 'govuk help', link: '/govuk/help', format: 'help_page')
+    commit_index
+    commit_index('govuk_test')
+
+    Search::FormatMigrator.stubs(:migrated_formats).returns(%w(help_page answer))
+
+    get "/search"
+
+    assert_equal(['govuk help'], parsed_response['results'].map { |r| r['title'] }.sort)
```
Comment (brenetic) [inline test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb]: Assert for `'govuk answer'`?

**Review by brenetic — COMMENTED** [review]
~~rubocop-781ff7c.html~~

One other comment otherwise MWA 🚀

File: test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb (hunk 3 above)
Comment (dwhenry) [reply test/integration/govuk_index/switch_on_formats_in_govuk_index_test.rb]: done..

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**